### PR TITLE
Add a blurb explaining the main synapse worker

### PR DESCRIPTION
**Pull request by turt2live**
While setting this up I ran into a couple issues that should hopefully be resolved by the documentation in this PR:
* Almost forgetting that the main synapse process is now a worker
* Getting errors for having `worker_*` config options (because copy/paste is a thing)

### CODE REVIEW

File: docs/workers.rst [hunk 1]
```diff
@@ -115,6 +115,18 @@ To manipulate a specific worker, you pass the -w option to synctl::
...
 
+After setting up your workers, you'll need to create a worker configuration for
+the main synapse process. That worker configuration should look like this:::
+
+    worker_app: synapse.app.homeserver
+    daemonize: true
+
+Be sure to keep this particular configuration limited as synapse may refuse to
+start if the regular ``worker_*`` options are given. The ``homeserver.yaml``
+configuration will be used to set up the main synapse process.
```
Comment (richvdh) [inline docs/workers.rst]: isn't homeserver.yaml used for all of the processes?

**Comment by turt2live** [comment]
The only reason it's at the end of the section is because everything before it talks about how workers are supposed to work. I'll update this to work it in more prominently though

File: docs/workers.rst (hunk 1 above)
Comment (turt2live) [reply docs/workers.rst]: It is, however the main worker is different in that it refuses to work with the standard worker_* options. I wanted to add a reminder that the homeserver.yaml is used instead.

Comment (richvdh) [reply docs/workers.rst]: I think your phrasing is a bit misleading then.

**Comment by turt2live** [comment]
@richvdh I've updated this. I threw in the documentation for https://github.com/matrix-org/synapse/issues/3018 because it's such a small change. I can make it its own PR if wanted though.

**Comment by richvdh** [comment]
\o/

**Pull request merged**